app/policies/reading_policy.rb

```diff
@@ -0,0 +1,29 @@
+# app/policies/reading_policy.rb
+class ReadingPolicy < ApplicationPolicy
+    class Scope < Scope
+      def resolve
+        scope.where(user_id: user.id)
+      end
+    end
+  
+    def create?
+        user.present?
```

Comment:
because we force authentication for readings, this will already be true.

Reply:
So, for this policy I think I was trying to override the create method in the parent class (application policy) which looks like this:

```ruby
    def create?
      false
    end
 ```
 
I know it's redundant given the authorization in the controller, but if I don't override it, it defaults to a false value, so I was trying to make it secure by verifying if a user was present in the reading policy. Is it better to just set the application policy to true? Not sure how best to gracefully and securely handle that.

Reply:
No, you are right for these 2 points:
- policies in application policy should **probably** always be set to false. This is the default behavior since all other policies inherit from it, and having it set to false means if we ever forget a policy, the worst case is nobody can access it.
- we do want to override individually for each policy and action.

Some nuance here that I think isn't 100% clear for you. Calling `authorize` in the controller only direct the controller to first check the policy available. If we don't call it, policies don't get applied. It does not in itself say who can do what, so it is not actually redundant. Does that make sense?

In this case, my comment was just to say if anyone logged in is allowed to perform this action, then you should probably just set it to `true` (which, in essence, is what you did by setting it to `user.present?` - but it's an added layer of abstraction)

Reply:
Ok so that's why it defaults to false if I don't specify a policy in the reading policy class because 'authorize' in the controller looks for policy to match to determine whether or not something is authorized. It doesn't determine it on its own. I think I got it. Thank you!